Tests/SentryTests/Performance/SentryTracerTests.swift

```diff
@@ -266,6 +298,87 @@ class SentryTracerTests: XCTestCase {
         XCTAssertEqual(1, transactionsWithAppStartMeasrurement.count)
         XCTAssertNil(SentrySDK.getAndResetAppStartMeasurement())
     }
+
+    #if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+
+    func testChangeStartTimeStamp_RemovesFramesMeasurement() {
+        let sut = fixture.getSut()
+        givenFrames(1, 1, 1)
+        sut.startTimestamp = Date()
+
+        sut.finish()
+
+        fixture.hub.group.wait()
+
+        XCTAssertEqual(1, fixture.hub.capturedEventsWithScopes.count)
+        let serializedTransaction = fixture.hub.capturedEventsWithScopes.first!.event.serialize()
+        XCTAssertNil(serializedTransaction["measurements"] as? [String: [String: Int]])
+    }
+
+    func testAddFramesMeasurement() {
+        let sut = fixture.getSut()
+
+        let slowFrames = 4
+        let frozenFrames = 1
+        let normalFrames = 100
+        let totalFrames = slowFrames + frozenFrames + normalFrames
+        givenFrames(slowFrames, frozenFrames, normalFrames)
+
+        sut.finish()
+
+        fixture.hub.group.wait()
+
+        XCTAssertEqual(1, fixture.hub.capturedEventsWithScopes.count)
+        let serializedTransaction = fixture.hub.capturedEventsWithScopes.first!.event.serialize()
+        let measurements = serializedTransaction["measurements"] as? [String: [String: Int]]
+
+        XCTAssertEqual([
+            "frames_total": ["value": totalFrames],
+            "frames_slow": ["value": slowFrames],
+            "frames_frozen": ["value": frozenFrames]
+        ], measurements)
+        XCTAssertNil(SentrySDK.getAndResetAppStartMeasurement())
+    }
+
+    private func givenFrames(_ slow: Int, _ frozen: Int, _ normal: Int) {
+        fixture.displayLinkWrapper.call()
+
+        // Slow frames
+        for _ in 0..<slow {
+            fixture.displayLinkWrapper.internalTimestamp += TestData.slowFrameThreshold + 0.001
+            fixture.displayLinkWrapper.call()
+        }
+
+        // Frozen frames
+        for _ in 0..<frozen {
+            fixture.displayLinkWrapper.internalTimestamp += TestData.frozenFrameThreshold + 0.001
+            fixture.displayLinkWrapper.call()
+        }
+
+        // Normal frames.
+        for _ in 0..<(normal - 1) {
+            fixture.displayLinkWrapper.internalTimestamp += TestData.slowFrameThreshold - 0.01
+            fixture.displayLinkWrapper.call()
+        }
+    }
+    #endif
+
+    private func addZeroFrames(measurements: [String: [String: Int]]) -> [String: [String: Int]] {
+        #if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+        var noFrames = [
+            "frames_total": ["value": 0],
+            "frames_slow": ["value": 0],
+            "frames_frozen": ["value": 0]
```

Comment:
not sure if I understood the test, does it mean that zeroed frames would be added as a measurement? I'd say we should not.

Reply:
Why not? If you have a transaction that is faster than one frame takes to render, this is useful information. Or you have a transaction taking 500 ms and there are no frames being rendered. Again could be useful.

Reply:
it'd make sense if one of them is not zeroed, but if all of them are zeroed, I don't see the point, how do we know if there were no frames at all because nothing to drawn or a frame took 500ms and the transaction was finished? this would not be possible to know with such numbers, we'd need to have one more field to tell, hey, we started drawing but so far we didn't account anything vs there was nothing to be drawn anyway

Reply:
@bruno-garcia what are your thoughts on this one ⬆️ ?